platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/signed/ReservedSignedState.java

```diff
@@ -116,6 +116,27 @@ public boolean isNotNull() {
         return new ReservedSignedState(signedState, reason);
     }
 
+    /**
+     * Try to get another reservation on the signed state. If the signed state is not closed, then a new reservation
+     * will be returned. If the signed state is closed, then null will be returned.
+     *
+     * @param reason a short description of why this SignedState is being reserved. Each location where a SignedState is
+     *               reserved should attempt to use a unique reason, as this makes debugging reservation bugs easier.
+     * @return a new wrapper around the state that holds a new reservation, or null if the signed state is closed
+     */
+    public @Nullable ReservedSignedState tryGetAndReserve(@NonNull final String reason) {
+        if (signedState == null) {
+            return new ReservedSignedState();
+        }
+        if (!signedState.tryIncrementReservationCount(reason, reservationId)) {
+            return null;
+        }
+        final ReservedSignedState reservedSignedState = new ReservedSignedState(signedState, reason);
+        // the constructor will increment the reservation count again, so we need to decrement it here
+        signedState.decrementReservationCount(reason, reservationId);
```

Comment:
Is it possible to do this without an extra increment+decrement? It seems like we lose a lot of benefit from being lockless if we do two extra atomic operations.

Reply:
fixed. I mentioned this in the PR comment then forgot about it.